ui/src/app/services/login.service.ts

```diff
@@ -0,0 +1,56 @@
+import { Injectable } from "@angular/core";
+import { ApiService } from "./api.service";
+import { BehaviorSubject, Subject, Subscriber, Subscription } from "rxjs";
+import { UserSessionService } from "./user-session.service";
```

Comment:
You don't appear to be using this anywhere in the LoginService. Remove unused imports.